### PR TITLE
Initial Import

### LINKED ISSUE #2: Setup go-da repository
Setup:

- [ ] LICENCE (Apache 2.0)
- [ ] NOTICE
- [ ] GitHub workflows for build and lint

### LINKED ISSUE #3: Initial code of DA interface
Draft initial version of DA interface code.

### LINKED ISSUE #4: Add dummy DA implementation for tests
Create as simple as possible DA implementation & corresponding tests.

### PR DESCRIPTION
**Pull request by tzdybal**
## Overview

This PR consists of initial CI configuration, DA interface definition, dummy DA implementation and unit tests.

Resolves #2 
Resolves #3
Resolves #4

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by nashqueue — CHANGES_REQUESTED** [review]
Thank you! Looks good, apart from a bit of a change

File: da.go [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+	// This method is synchronous. Upon successful submission to Data Availability layer, it returns ID identifying blob
+	// in DA and Proof of inclusion.
 	Submit(blob Blob) (ID, Proof, error)
+
+	// Validate validates Commitment against Proof. This should be possible without retrieving Blob.
 	Validate(commit Commitment, proof Proof) (bool, error)
 }
 
+// BatchDA is a batched version of DA interface.
 type BatchDA interface {
```
Comment (MSevey) [inline da.go:25]: what's the benefit of having both a `BatchDA` and `DA` vs just having the `DA` accept slices? You can still submit a single item to the `BatchDA`, so it feels like unnecessary duplication.

**Review by MSevey — COMMENTED** [review]
should also include this in the golangci

```yml

linters-settings:
  revive:
    rules:
      - name: exported
        arguments:
          - disableStutteringCheck

issues:
  include:
    - EXC0012  # EXC0012 revive: Annoying issue about not having a comment. The rare codebase has such comments
    - EXC0014  # EXC0014 revive: Annoying issue about not having a comment. The rare codebase has such comments
   ```

ref: https://github.com/rollkit/rollkit/issues/587

**Review by gupadhyaya — COMMENTED** [review]
LGTM apart from the validate parameter being ID instead of Commitment

File: da.go (hunk 1 above)
Comment (gupadhyaya) [reply da.go:25]: agree. just `type DA interface {` that supports array should suffice.

Comment (tuxcanfly) [reply da.go:25]: The batch operations can be atomic i.e. either all blobs are submitted successfully or none is. This way we can rollback and try again with all pending blobs without having to worry about which ones went through and which ones didn't.

Comment (tzdybal) [reply da.go]: Updated to `[]` everywhere.

**Review by MSevey — APPROVED** [review]

**Review by tuxcanfly — APPROVED** [review]
OK.

**Review by nashqueue — APPROVED** [review]

**Pull request merged**